src/evict.c

```diff
@@ -729,6 +740,14 @@ int performEvictions(void) {
         }
     }
 
+    serverAssert(server.core_propagates); /* This function should not be re-entrant */
```

Comment:
how this line makes this function non-reentrant? server.core_propagates can be change by other code.